## Exploratory Data Analysis for Microsoft's New Movie Studio

## Overview

In this project, we will perform exploratory data analysis (EDA) to generate insights for Microsoft, 
helping them decide what type of films to create for their new movie studio. We will use various datasets 
related to movies, analyze trends, and provide actionable recommendations based on our findings.


## Business Problem

Microsoft sees all the big companies creating original video content and they want to get in on the fun. 
They have decided to create a new movie studio but lack knowledge about creating movies. Our task is to explore 
what types of films are currently doing the best at the box office and translate those findings into actionable insights 
for Microsoft's new movie studio.

## Objectives

The objectives of this project are to:
1. Identify the top-performing movie genres at the box office by analyzing historical box office data.
2. Determine the relationship between movie ratings and box office success to recommend the ideal quality standard for new films.
3. Analyze production budgets and their impact on box office returns to suggest optimal budget ranges for new movie projects.

## Loading Data

In [106]:
import pandas as pd 
import numpy as np
import sqlite3
import matplotlib.pyplot as plt
import seaborn as sns

In [107]:
bom_movie_gross = pd.read_csv('data/bom.movie_gross.csv')
rt_movie_info = pd.read_csv('data/rt.movie_info.tsv', sep='\t')
rt_reviews = pd.read_csv('data/rt.reviews.tsv', sep='\t', encoding='ISO-8859-1')
tmdb_movies = pd.read_csv('data/tmdb.movies.csv')
tn_movie_budgets = pd.read_csv('data/tn.movie_budgets.csv')

conn = sqlite3.connect('data/im.db')
imdb_movie_basics = pd.read_sql_query("SELECT * FROM movie_basics", conn)
imdb_movie_ratings = pd.read_sql_query("SELECT * FROM movie_ratings", conn)

Displaying the first few rows of each dataframe.

In [108]:

print("Movie Gross Data:")
print(bom_movie_gross.head())
print("-------------------------------------------------------------")

print("Movie Info Data:")
print(rt_movie_info.head())
print("-------------------------------------------------------------")

print("Reviews Data:")
print(rt_reviews.head())
print("-------------------------------------------------------------")

print("TMDB Movies Data:")
print(tmdb_movies.head())
print("-------------------------------------------------------------")

print("Movie Budgets Data:")
print(tn_movie_budgets.head())
print("-------------------------------------------------------------")

print("Movie Basics Data:")
print(imdb_movie_basics.head())
print("-------------------------------------------------------------")

print("Movie Ratings Data:")
print(imdb_movie_ratings.head())

Movie Gross Data:
                                         title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010  
-------------------------------------------------------------
Movie Info Data:
   id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous s

Upon initial examination of the dataframes, the following datasets were selected as they contain the necessary information to meet the project objectives:

- bom_movie_gross
- tn_movie_budgets
- imdb_movie_basics
- imdb_movie_ratings

#### Inspecting the selected datasets 

#### 1. bom_movie_gross

In [109]:
#Examining the total number of rows, columns, non-null values and datatypes in the dataframe
bom_movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [110]:
# Checking for null values
bom_movie_gross.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [111]:
#checking for duplicetes
bom_movie_gross.duplicated().sum()

0

#### 2. tn_movie_budgets

In [112]:
#Examining the total number of rows, columns, non-null values and datatypes in the dataframe
tn_movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [113]:
# Checking for null values
tn_movie_budgets.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

In [114]:
#checking for duplicetes
tn_movie_budgets.duplicated().sum()

0

#### 3. imdb_movie_basics

In [115]:
#Examining the total number of rows, columns, non-null values and datatypes in the dataframe
imdb_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


In [116]:
# Checking for null values
imdb_movie_basics.isna().sum()

movie_id               0
primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

In [117]:
#checking for duplicetes
imdb_movie_basics.duplicated().sum()

0

#### 4. imdb_movie_ratings

In [118]:
#Examining the total number of rows, columns, non-null values and datatypes in the dataframe
imdb_movie_ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   movie_id       73856 non-null  object 
 1   averagerating  73856 non-null  float64
 2   numvotes       73856 non-null  int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 1.7+ MB


In [119]:
# Checking for null values
imdb_movie_ratings.isna().sum()

movie_id         0
averagerating    0
numvotes         0
dtype: int64

In [120]:
#checking for duplicetes
imdb_movie_ratings.duplicated().sum()

0

Upon inspecting the datasets, the bom_movie_gross dataframe was dropped since its data is available in other datasets. Additionally, the foreign_gross column had 1,350 missing values out of a total of 3,387 rows.

## Data Cleaning

### Missing Values

##### imdb_movie_basics

In [121]:
#checking the percentages of the missing data



There are 21 missing values in the original_title column, 31,739 missing values in the runtime_minutes column, and 5,408 missing values in the genres column. These were dropped due to their relatively small number compared to the entire dataset.

In [122]:
imdb_movie_basics.dropna(inplace=True)

In [123]:
imdb_movie_basics.isna().sum()

movie_id           0
primary_title      0
original_title     0
start_year         0
runtime_minutes    0
genres             0
dtype: int64

In [124]:
#merging the imdb dataframes
imdb_merged = pd.merge(imdb_movie_basics, imdb_movie_ratings, how='inner', on='movie_id')

#checking the fisrt five rows of the merged dataframe
imdb_merged.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0100275         The Wandering Soap Opera       La Telenovela Errante   
4  tt0137204                  Joe Finds Grace             Joe Finds Grace   

   start_year  runtime_minutes                      genres  averagerating  \
0        2013            175.0          Action,Crime,Drama            7.0   
1        2019            114.0             Biography,Drama            7.2   
2        2018            122.0                       Drama            6.9   
3        2017             80.0        Comedy,Drama,Fantasy            6.5   
4        2017             83.0  Adventure,Animation,Comedy            8.1   

   numvotes  
0        77  
1        43  
2      4517  
3       119  
4       263

In [125]:
#checking the shape of the merged object
imdb_merged.shape

(65720, 8)

After cleaning the data, the remaining dataframes containing the cleaned data are:
- tn_movie_budgets
- imdb_merged

## Creating New Variables for Analysis

### Analyzing revenue

In [126]:
#converting production_budget, domestic_gross, and worldwide_gross columns in tn_movie_budgets to numeric values
tn_movie_budgets['production_budget'] = tn_movie_budgets['production_budget'].replace('[\$,]', '', regex=True).astype(float)
tn_movie_budgets['domestic_gross'] = tn_movie_budgets['domestic_gross'].replace('[\$,]', '', regex=True).astype(float)
tn_movie_budgets['worldwide_gross'] = tn_movie_budgets['worldwide_gross'].replace('[\$,]', '', regex=True).astype(float)

#calculating profit
tn_movie_budgets['domestic_profit'] = tn_movie_budgets['domestic_gross'] - tn_movie_budgets['production_budget']
tn_movie_budgets['worldwide_profit'] = tn_movie_budgets['worldwide_gross'] - tn_movie_budgets['production_budget']

#calcultaing return on investment
tn_movie_budgets['domestic_roi'] = tn_movie_budgets['domestic_profit'] / tn_movie_budgets['production_budget']
tn_movie_budgets['worldwide_roi'] = tn_movie_budgets['worldwide_profit'] / tn_movie_budgets['production_budget']

#Extracting year from release date
tn_movie_budgets['release_year'] = pd.to_datetime(tn_movie_budgets['release_date']).dt.year

tn_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  domestic_profit  \
0        425000000.0     760507625.0     2.776345e+09      335507625.0   
1        410600000.0     241063875.0     1.045664e+09     -169536125.0   
2        350000000.0      42762350.0     1.497624e+08     -307237650.0   
3        330600000.0     459005868.0     1.403014e+09      128405868.0   
4        317000000.0     620181382.0     1.316722e+09      303181382.0   

   worldwide_profit  domestic_roi  worldwide_roi  release_year  
0      2.351345e+09      0.789430       5.532577          2009  
1      6.350639e+08     -0.412899       1.546673          2011  
2     -2.002376e+08     -0.877822      -0.572108          2019  
3      1.072414e+09      0.388403       3.243841          2015  
4      9.997217e+08      0.956408       3.153696          2017

### Analyzing Genres

In [129]:
#using one-hot encoding to seperate the genres

genres = imdb_merged['genres'].str.get_dummies(sep=',')
imdb_merged = pd.concat([imdb_merged, genres], axis=1)

In [130]:
imdb_merged.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0100275         The Wandering Soap Opera       La Telenovela Errante   
4  tt0137204                  Joe Finds Grace             Joe Finds Grace   

   start_year  runtime_minutes                      genres  averagerating  \
0        2013            175.0          Action,Crime,Drama            7.0   
1        2019            114.0             Biography,Drama            7.2   
2        2018            122.0                       Drama            6.9   
3        2017             80.0        Comedy,Drama,Fantasy            6.5   
4        2017             83.0  Adventure,Animation,Comedy            8.1   

   numvotes  Action  Adult  ...  Mystery  News  Reality-TV  Romance  Sci-Fi  \
0        77       1      0  ...        0     0           0        0       0   
1        43       0      0  ...        0     0           0        0       0   
2      4517       0      0  ...        0     0           0        0       0   
3       119       0      0  ...        0     0           0        0       0   
4       263       0      0  ...        0     0           0        0       0   

   Short  Sport  Thriller  War  Western  
0      0      0         0    0        0  
1      0      0         0    0        0  
2      0      0         0    0        0  
3      0      0         0    0        0  
4      0      0         0    0        0  

[5 rows x 34 columns]

In [131]:
#merging the tn_movie_budgets and the imdb_merged dataframe

tn_movie_budgets = tn_movie_budgets.rename(columns={'movie': 'primary_title'})
combined_data = pd.merge(tn_movie_budgets, imdb_merged, on='primary_title', how='inner')


In [132]:
combined_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2752 entries, 0 to 2751
Data columns (total 44 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 2752 non-null   int64  
 1   release_date       2752 non-null   object 
 2   primary_title      2752 non-null   object 
 3   production_budget  2752 non-null   float64
 4   domestic_gross     2752 non-null   float64
 5   worldwide_gross    2752 non-null   float64
 6   domestic_profit    2752 non-null   float64
 7   worldwide_profit   2752 non-null   float64
 8   domestic_roi       2752 non-null   float64
 9   worldwide_roi      2752 non-null   float64
 10  release_year       2752 non-null   int64  
 11  movie_id           2752 non-null   object 
 12  original_title     2752 non-null   object 
 13  start_year         2752 non-null   int64  
 14  runtime_minutes    2752 non-null   float64
 15  genres             2752 non-null   object 
 16  averagerating      2752 

## Exploratory Data Analysis (EDA)